cmd/openshift-install/create.go

```diff
@@ -179,17 +180,28 @@ func destroyBootstrap(ctx context.Context, directory string) (err error) {
 
 	discovery := client.Discovery()
 
-	apiContext, cancel := context.WithTimeout(ctx, 30*time.Minute)
+	apiContext, cancel := context.WithTimeout(ctx, bootstrapTimeout)
 	defer cancel()
+	previousError := ""
+	// Only show error messages after 30s have elapsed; before that we know
+	// that even in the fastest setups that machines are going to be
+	// booting, running Ignition, etc.
+	beginVerboseTime := time.Now().Add(30 * time.Second)
 	wait.Until(func() {
 		version, err := discovery.ServerVersion()
 		if err == nil {
 			logrus.Infof("API %s up", version)
 			cancel()
 		} else {
-			logrus.Debugf("API not up yet: %s", err)
+			errText := err.Error()
+			if previousError != errText && time.Now().After(beginVerboseTime) {
+				logrus.Infof("API not up yet: %s", errText)
```

Comment:
This seems too verbose. The average user is going to see a terminal full of these before their cluster comes up, which I think is just going to confuse or annoy. Remember, the customer here is AWS users; not libvirt.

Reply:
Once every 5 minutes?

Reply:
The period is 5 seconds below (unless `ServerVersion()` has some built-in retry).

Reply:
Yeah, it's 5s right now...I'm negotiating here for a different value :smile:

Reply:
> Every five minutes seems reasonable, but it won't be as simple as adjusting the period below (that will cause us to delay up to five minutes after the API has been up before we check again).

You can resolve this with high-frequency polling and then downsampling before printing logs (so only log on changes or on every 15th unchanged poll).  For an implementation, see [here][1].  That allows to you have your fast-polling responsiveness and your low-spam logs, at the cost of a slightly-more-complicated implementation.  I'm fine paying the implementation cost, but...

> Given this added complexity and the fact that we are planning on adding more installation milestones (e.g. "we booted the bootstrap node", "we created the temporary control plane")...

I'm also fine waiting for this.

[1]: https://github.com/openshift/installer/pull/778#discussion_r238535608